core/src/main/java/org/infinispan/remoting/transport/Transport.java

```diff
@@ -154,4 +155,20 @@
     * protocol stack.
     */
    void checkTotalOrderSupported();
+
+   void addXSiteViewListener();
```

Comment:
I don't think this should be exposed to the public and the ```Transport``` implementation should register automatically a listener.